File: find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        vector<int>res;
        int max,i,j;
        for (i=0;i<nums.size();i++){
            if (nums[i]==target){res.push_back(i);break;}
        }
        for (j=0;j<nums.size();j++){
            if (nums[j] == target)max=j;
        }
        if (i==nums.size() and j== nums.size()){res.push_back(-1);res.push_back(-1);}
        else res.push_back(max);
        return res;
    }
};

File: find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] arr, int target) {
         int []ans={-1,-1};
       //check for first and last occurance
       ans[0]=search(arr,target,true);
       ans[1]=search(arr,target,false);
        return ans;
    }
     int search(int []arr,int target,boolean findFirst){
        int low =0;
        int high= arr.length-1,ans=-1;
        while(low<=high){
            int mid = low + (high-low)/2;
            if (target<arr[mid]){
                high=mid-1;
            }
            else if (target > arr[mid])low = mid+1;
            else {
                ans = mid;
                if (findFirst){ high= mid-1;}
                else{ low = mid+1;}
            }
            
        }
        return ans;
    }
}